kpm/lib/kpm/nexus_helper/nexus_api_calls_v2.rb

```diff
@@ -23,71 +24,81 @@ def message
 
     # This is an extract and slim down of functions needed from nexus_cli to maintain the response expected by the base_artifact.
     class NexusApiCallsV2
-      PULL_ARTIFACT_ENDPOINT = 'service/local/artifact/maven/redirect'
-      GET_ARTIFACT_INFO_ENDPOINT = 'service/local/artifact/maven/resolve'
-      SEARCH_FOR_ARTIFACT_ENDPOINT = 'service/local/data_index'
+      PULL_ARTIFACT_ENDPOINT = '/service/local/artifact/maven/redirect'
+      GET_ARTIFACT_INFO_ENDPOINT = '/service/local/artifact/maven/resolve'
+      SEARCH_FOR_ARTIFACT_ENDPOINT = '/service/local/data_index'
+
+      READ_TIMEOUT_DEFAULT = 60000
+      OPEN_TIMEOUT_DEFAULT = 60000
+
+      ERROR_MESSAGE_404 = 'The artifact you requested information for could not be found. Please ensure it exists inside the Nexus.'
+      ERROR_MESSAGE_503 = 'Could not connect to Nexus. Please ensure the url you are using is reachable.'
 
       attr_reader :version
       attr_reader :configuration
       attr_reader :ssl_verify
+      attr_accessor :logger
 
-      def initialize(configuration, ssl_verify)
+      def initialize(configuration, ssl_verify, logger)
         @configuration = configuration
         @ssl_verify = ssl_verify
+        @logger = logger
       end
 
       def search_for_artifacts(coordinates)
-        http_client = get_client
-        url = get_url(SEARCH_FOR_ARTIFACT_ENDPOINT, configuration)
-        group_id, artifact_id = coordinates.split(":")
-        response = http_client.get(url, :query => {:g => group_id, :a => artifact_id})
-
-        case response.status
-          when 200
-            return response.content
+        logger.debug 'Entered - Search for artifact'
```

Comment:
Better to merge these two lines in one (`Entered - Search for artifact, coordinates: #{coordinates}"`), as it makes it easier to grep logs.